ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlServerMain.java

```diff
@@ -99,4 +104,28 @@ private static Executable createExecutable(
     builder.put(StreamsConfig.APPLICATION_ID_CONFIG, KSQL_REST_SERVER_DEFAULT_APP_ID);
     return builder.build();
   }
+
+  // package-private for test purpose
+  static void enforceStreamStateDirAvailability(final File streamsStateDir) {
+    if (!streamsStateDir.exists() || !streamsStateDir.isDirectory()) {
+      throw new KsqlServerException("The kafka streams state directory does not exist: "
+          + streamsStateDir.getPath()
+          + "\n Make sure the directory exists and is writable for KSQL server."
+      );
+    }
+    if (!streamsStateDir.canWrite()) {
+      throw new KsqlServerException("The kafka streams state directory is not writable "
+          + "for KSQL server: "
+          + streamsStateDir.getPath()
+          + "\n Make sure KSQL server has write access to this directory or change it to a writable"
+          + " one by setting `ksql.streams.state.dir` config in the properties file."
+      );
+    }
+  }
+
+  private static StreamsConfig getSampleStreamsConfig() {
+    return new StreamsConfig(
```

Comment:
Creating a dummy config just to get the default value for a single key feels very weird. Can we not do something like `StreamsConfig.configDef().defaultValues().get(StreamsConfig.STATE_DIR_CONFIG)` instead?

Reply:
You are right, this one is much better. Updated.